### DISCUSSION TITLE
Implement a hook api with new essentials safe /rtp feature

### DISCUSSION THREAD

**Discussion by LoSCraft-Network** [GriefPrevention/GriefPrevention#878] [Ideas]
The newest update on essentials brought an /rtp command, to random teleport players.

Talking to essentials team, I was told that their "safe teleport" is only safe as in a manner of lava, water, holes and things like that.

They said any other plugin-side protection has to be added through their API.

It would be nice to add GriefPrevention to that, making RTP not Randomly teleport Players inside claimed area.

**Comment by RoboMWM** [comment]
Anyone's free to PR GP to that, but personally I have a [random tp plugin](https://www.spigotmc.org/resources/wilderness-tp.22853/) that avoids all areas the player does not have break (or build access if configured to check that) access in.

**Comment by TomLewis** [comment]
+1 for Wilderness-Tp

**Comment by LoSCraft-Network** [comment]
I will try out the one you personally use!

But even tho it would be nice to have it built in so it already works with what we already have, no need for an extra plugin
